crates/wasm_utils/src/api_serialization/keystore.rs

```diff
@@ -21,13 +21,32 @@ pub struct KeystoreDeriveSeedArgs {
     pub dst_id: String,
     pub context: String,
     pub index: u64,
+    pub seed_type: SeedType,
 }
 
 #[derive(Deserialize, Clone, PartialEq, Eq, Hash, Debug, Serialize, DefaultJson)]
 pub enum KeyType {
     Signing,
     Encrypting,
 }
+/// Enum of all the types of seeds
+#[derive(Serialize, Deserialize, Clone, Debug, Hash, PartialEq, Eq)]
+pub enum SeedType {
+    /// Root / Master seed
+    Root,
+    /// Revocation seed
+    Revocation,
```

Comment:
Need to add `Auth` type here
```suggestion
    Revocation,
    /// Auth seed
    Auth,
```